kratos/processes/assign_scalar_input_to_entities_process.h

```diff
@@ -0,0 +1,352 @@
+    ///@name  Enum's
+    ///@{
+
+    /**
+     * @brief This enum helps us to identify the algorithm considered
+     */
+    enum class Algorithm {
+        NEAREST_NEIGHBOUR = 0
+    };
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor
+     * @param rModelPart The model part to be set
+     * @param rParameters The configuration parameters
+     */
+    AssignScalarInputToEntitiesProcess(
+        ModelPart& rModelPart,
+        Parameters rParameters
+        );
+
+    /// Destructor.
+    ~AssignScalarInputToEntitiesProcess() override {}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This function will be executed at every time step BEFORE performing the solve phase
+     */
+    void ExecuteInitializeSolutionStep() override;
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     */
+    const Parameters GetDefaultParameters() const override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "AssignScalarInputToEntitiesProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "AssignScalarInputToEntitiesProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+    ///@}
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+    ///@}
+    ///@name Protected Operators
+    ///@{
+    ///@}
+    ///@name Protected Operations
+    ///@{
+    ///@}
+    ///@name Protected  Access
+    ///@{
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+    ///@}
+
+private:
+
+    ///@name Static Member Variables
+    ///@{
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;                                                  /// The model part where to assign the values
+
+    const Variable<double>* mpVariable = nullptr;                            /// The pointer of the variable
+
+    ResultDatabase mDatabase;                                                /// The database containing the information to assign the values
+
+    std::vector<std::unordered_map<IndexType, double>> mWeightExtrapolation; /// This maps specifies the extrapolation weights for each entity
+
+    std::vector<array_1d<double, 3>> mCoordinates;                           /// The coordinates of the database
+
+    Algorithm mAlgorithm = Algorithm::NEAREST_NEIGHBOUR;                     /// The algorithm considered
```

Comment:
Shouldn’t this fit directly? If not then one could use the mappers to fit the results 
I don’t like reimplementing the same functionality

Reply:
It is not a mapping (unless you create an auxiliar mesh), you give a coordinates and it assigns values according to that into the real mesh

Reply:
Hm but why not creating a temp aux mesh?
Then you can use all the available mappers if the mesh is not matching 

This way would even work in mpi (using the mappers from the mapping App)

Reply:
I was thinking about this in a future PR, this is the first version

Reply:
For the tasks we need to perform now it serves perfectly the way it is.